core-web/apps/dotcms-ui/src/app/view/components/dot-crumbtrail/service/dot-crumbtrail.service.ts

```diff
@@ -123,6 +123,24 @@ export class DotCrumbtrailService {
         const portletId = replaceSectionsMap[sections[0]] || sections[0];
 
         return this.getMenuLabel(portletId).pipe(
+            switchMap((crumbTrail: DotCrumb[]) =>
+                // If it is edit page
+                sections[0] == 'edit-page'
+                    ? // We get the breadcrumb for pages
+                      this.getMenuLabel('pages').pipe(
+                          map(
+                              (pagesCrumbTrail: DotCrumb[]) =>
+                                  // If the pages portlet exists in the menu
+                                  pagesCrumbTrail.length
+                                      ? pagesCrumbTrail.filter(
+                                            (value) => !value.url.includes('site-browser')
+                                        ) // We remove the Site link and return the breadcrumb
+                                      : crumbTrail // Otherwise we return the original breadcrumb
+                          )
+                      )
+                    : // If it's not edit pages, we return the original breadcrumb
+                      of(crumbTrail)
+            ),
```

Comment:
This a suggestion to make the code more readable.

```suggestion
        const portletId = replaceSectionsMap[sections[0]] || sections[0];
        const isEditPage = sections && sections[0] === 'edit-page';

        return this.getMenuLabel(portletId).pipe(
            switchMap((crumbTrail: DotCrumb[]) => {
                // if is edit page we need to get the pages crumbtrail
                return isEditPage ? this.getPagesCrumbTrail(crumbTrail) : of(crumbTrail);
            }),
```

And the `getPagesCrumbTrail` will be:

```ts
    /**
     * Get the pages crumbtrail.
     * Alternate crumbtrail is used when the page portlet is not enabled.
     *
     * @private
     * @param {DotCrumb[]} alternateCrumbTrail
     * @return {*}  {Observable<DotCrumb[]>}
     * @memberof DotCrumbtrailService
     */
    private getPagesCrumbTrail(alternateCrumbTrail: DotCrumb[] = []): Observable<DotCrumb[]> {
        return this.getMenuLabel('pages').pipe(
            map((pagesCrumbTrail: DotCrumb[]) => {
                // Remove the site-browser from the pages crumbtrail
                const crumbTail = pagesCrumbTrail?.filter(
                    (value) => !value.url.includes('site-browser')
                );

                return crumbTail.length ? crumbTail : alternateCrumbTrail;
            })
        );
    }
```